sdk/appcenter/src/main/java/com/microsoft/appcenter/AppCenter.java

```diff
@@ -365,7 +387,23 @@ public static void startFromLibrary(Context context, Class<? extends AppCenterSe
      * @see AppCenterFuture
      */
     public static AppCenterFuture<UUID> getInstallId() {
-        return getInstance().getInstanceInstallId();
+        return getInstance().getInstanceInstallIdAsync();
+    }
+
+    /**
+     * Set the SQLite database storage size. Returns true if the operation succeeded. If the new size
+     * is smaller than the previous size (database is shrinking) and the capacity is greater than
+     * the new size, then the operation will fail and a warning will be emitted. Can only be called
+     * once per app lifetime and only before AppCenter.start(...).
+     * <p>
+     * If the size is not a multiple of 4096 bytes, the next multiple of 4096 is used as the new maximum size.
+     *
+     * @param storageSizeInBytes New size for the SQLite db in bytes.
+     * @return Future with true result if succeeded, otherwise future with false result.
+     */
+    @SuppressWarnings("WeakerAccess") // TODO remove annotation when updating demo app for release.
+    public static AppCenterFuture<Boolean> setMaxStorageSize(long storageSizeInBytes) {
```

Comment:
Doesn't the spec indicate that we need to use a callback?

Reply:
If I understand correctly, a future is analagous to a callback in that we're saving it here, and completing either immediately (if there is some parameter/timing violation) or saving and completing it [here](https://github.com/Microsoft/AppCenter-SDK-Android/pull/796/files#diff-918508e799dc541fd29d897160e9b0a5R749)